python/mspasspy/algorithms/ml/arrival.py

```diff
@@ -0,0 +1,80 @@
+import numpy as np
+import seisbench.models as sbm
+
+from seisbench.models.base import WaveformModel
+from obspy.taup import TauPyModel
+from obspy.geodetics import gps2dist_azimuth, kilometers2degrees
+from mspasspy.ccore.seismic import TimeSeries, TimeSeriesEnsemble
+from mspasspy.ccore.algorithms.basic import TimeWindow
+from mspasspy.algorithms.window import WindowData
+
+
+def annotate_arrival_time(
+    timeseries: TimeSeries,
+    threshold=0.2,
+    time_window: TimeWindow = None,
+    model: WaveformModel = None,
+    model_args: dict = None,
+):
+    """
+    Predict the arrival time of the P wave using the provided seisbench WaveformModel.
+    The arrival time will saved as metadata in the input TimeSeries object and can be accessed using \
+        the key 'p_wave_picks'. Time is stored in the relative format using the start time of the TimeSeries.
+
+    :param timeseries: The time series data to predict the arrival time.
+    :param threshold: The probability threshold (0-1) to filter p-wave picks. Default value is 0.2.
+    :param time_window: The relative time window to filter the predicted arrival time. If not provided, the whole time series will be used.
+    :param model: The model used to predict the arrival time.
+    :param model_args: arguments to initialize a new model if not provided
+    :type timeseries: mspasspy.ccore.seismic.TimeSeries
+    :type threshold: float
+    :type time_window: mspasspy.ccore.algorithms.basic.TimeWindow
+    :type model: seisbench.models.base.WaveformModel
+    :type model_args: dict
+    """
+
+    # Check the input arguments
+    if not 0 <= threshold <= 1:
+        raise ValueError("Threshold must be between 0 and 1")
```

Comment:
instead of raising an error, maybe this should be a warning instead and we will set the threshold to 0 or 1 if the range is exceeded.